### PR TITLE
[Auditv2] e2e test fixes (for master)

### LINKED ISSUE #33571: Update e2e tests to be id-agnostic

### PR DESCRIPTION
**Pull request by iethree**
Finally Closes https://github.com/metabase/metabase/issues/33571

### Description

Cherry-picking the test fixes from https://github.com/metabase/metabase/pull/34006 into master to make sure they work here too.

Mostly, this makes miscellaneous tests id-agnostic, but it also makes some tests work with variable numbers of collections and other little idiosyncrasies that auditv2 will introduce when it merges to master.

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: e2e/test/scenarios/sharing/reproductions.cy.spec.js
Comment (iethree) [inline e2e/test/scenarios/sharing/reproductions.cy.spec.js:1]: this combines 5 separate tests into a single file.

File: e2e/test/scenarios/binning/binning-reproductions.cy.spec.js [hunk 1]
```diff
@@ -172,7 +172,7 @@ describe("binning related reproductions", () => {
       cy.findByRole("option", { name: "CREATED_AT" }).click();
     });
 
-    getNotebookStep("summarize").findByText("Question 5 → Created At: Month");
+    getNotebookStep("summarize").findByText(/Question \d+ → Created At: Month/);
```
Comment (npfitz) [inline e2e/test/scenarios/binning/binning-reproductions.cy.spec.js:175]: 🔥

File: e2e/test/scenarios/dashboard-cards/dashboard-drill.cy.spec.js [hunk 2]
```diff
@@ -919,14 +923,13 @@ describe("scenarios > dashboard > dashboard drill", () => {
...
-      cy.findByText("110.93");
-      // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
-      cy.findByText("52.72").should("not.exist");
-      // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
-      cy.findByText("Showing 2 rows");
+      cy.findByTestId("TableInteractive-root").within(() => {
+        cy.findByText("37.65");
+        cy.findByText("110.93");
+        cy.findByText("52.72").should("not.exist");
+      });
```
Comment (npfitz) [inline e2e/test/scenarios/dashboard-cards/dashboard-drill.cy.spec.js:930]: 👏 Love it

**Review by npfitz — APPROVED** [review]
Everything looks good to me 👍

**Review by nemanjaglumac — APPROVED** [review]
CI is green, roses are red.
You are awesome!

**Pull request merged**